src/utils/use-debounce.js

```diff
@@ -0,0 +1,11 @@
+import { useMemo } from 'react';
+import { debounce } from '@mui/material/utils';
+
+const useDebounce = (updateParameter, delay) => {
```

Comment:
rename `updateParameter` to `func` because it's very generic.
It is as much generic that you should add this custom hook in commons-ui I think